app/main/views/services.py

```diff
@@ -671,7 +697,68 @@ def edit_service_submission(framework_slug, lot_slug, service_id, section_id, qu
 
 
 @main.route('/frameworks/<framework_slug>/submissions/<lot_slug>/<service_id>/remove/<section_id>/<question_slug>',
-            methods=['GET', 'POST'])
+            methods=['GET'])
+@login_required
+@EnsureApplicationCompanyDetailsHaveBeenConfirmed(data_api_client)
+@return_404_if_applications_closed(lambda: data_api_client)
+def remove_subsection_warning(framework_slug, lot_slug, service_id, section_id, question_slug):
+    framework, lot = get_framework_and_lot_or_404(data_api_client, framework_slug, lot_slug, allowed_statuses=['open'])
+
+    # Suppliers must have registered interest in a framework before they can edit draft services
+    if not get_supplier_framework_info(data_api_client, framework_slug):
+        abort(404)
```

Comment:
Why do we need both this check and `@EnsureApplicationCompanyDetailsHaveBeenConfirmed`?

Reply:
Oh - I see. This is a pre-existing thing. I think the `if not get_supplier_framework_info(data_api_client, framework_slug):` check is redundant and can be removed. I'm going to raise a PR to remove it from the existing endpoints.